libraries/bot-ai-qna/src/main/java/com/microsoft/bot/ai/qna/QnAMakerRecognizer.java

```diff
@@ -0,0 +1,552 @@
+        }
+
+        String epKey = this.endpointKey;
+        String hn = this.hostName;
+        String kbId = this.knowledgeBaseId;
+        Boolean logPersonalInfo = this.logPersonalInformation;
+        QnAMakerEndpoint endpoint = new QnAMakerEndpoint() {
+            {
+                setEndpointKey(epKey);
+                setHost(hn);
+                setKnowledgeBaseId(kbId);
+            }
+        };
+
+        return CompletableFuture.completedFuture(
+                new QnAMaker(endpoint, new QnAMakerOptions(), this.getTelemetryClient(), logPersonalInfo));
+    }
+
+    /**
+     * Uses the RecognizerResult to create a list of properties to be included when
+     * tracking the result in telemetry.
+     *
+     * @param recognizerResult    Recognizer Result.
+     * @param telemetryProperties A list of properties to append or override the
+     *                            properties created using the RecognizerResult.
+     * @param dialogContext       Dialog Context.
+     * @return A dictionary that can be included when calling the TrackEvent method
+     *         on the TelemetryClient.
+     */
+    @Override
+    protected Map<String, String> fillRecognizerResultTelemetryProperties(RecognizerResult recognizerResult,
+            Map<String, String> telemetryProperties, @Nullable DialogContext dialogContext) {
+        if (dialogContext == null) {
+            throw new IllegalArgumentException(
+                    "dialogContext: DialogContext needed for state in AdaptiveRecognizer.FillRecognizerResultTelemetryProperties method.");
+        }
+
+        Map<String, String> properties = new HashMap<String, String>() {
+            {
+                put("TopIntent",
+                        !recognizerResult.getIntents().isEmpty()
+                                ? (String) recognizerResult.getIntents().keySet().toArray()[0]
+                                : null);
+                put("TopIntentScore",
+                        !recognizerResult.getIntents().isEmpty()
+                                ? Double.toString(
+                                        ((IntentScore) recognizerResult.getIntents().values().toArray()[0]).getScore())
+                                : null);
+                put("Intents",
+                        !recognizerResult.getIntents().isEmpty()
+                                ? Serialization.toStringSilent(recognizerResult.getIntents())
+                                : null);
+                put("Entities",
+                        recognizerResult.getEntities() != null
+                                ? Serialization.toStringSilent(recognizerResult.getEntities())
+                                : null);
+                put("AdditionalProperties",
+                        !recognizerResult.getProperties().isEmpty()
+                                ? Serialization.toStringSilent(recognizerResult.getProperties())
+                                : null);
+            }
+        };
+
```

Comment:
Here is missing this [part](https://github.com/microsoft/botbuilder-dotnet/blob/main/libraries/Microsoft.Bot.Builder.AI.QnA/QnAMakerRecognizer.cs#L329), is that correct?

Reply:
+1. I'll update this